### PR TITLE
Adding quail

### PR DESCRIPTION
**Pull request by yzpang**
Adding quail ([project page](https://text-machine-lab.github.io/blog/2020/quail/)).

Used our version of processed data. May need to add data preprocessing scripts later, depending on discussion in the meeting today. 

Using this [colab](https://colab.research.google.com/drive/1oKnMFzVJnhTt22pIJ_OQhFRehjQ1iw2a?usp=sharing), based on RoBERTa-base, obtained dev set accuracy of 0.468. On par with [existing performance](http://text-machine.cs.uml.edu/lab2/projects/quail/) (though on test set).

### CODE REVIEW

**Review by jeswan — CHANGES_REQUESTED** [review]
Thanks for the PR! Please add the task to `guides/tasks/adding_tasks.md`. Although not officially part of the adding a task requirements, it would be useful to add this task to the data downloader since the task is part of [`datasets` ](https://huggingface.co/nlp/viewer/?dataset=quail).

**Comment by yzpang** [comment]
Thanks! Done. I haven't implemented downloader for quail given that we're using a very different format of the dataset. I might submit more PRs later to add extensive data preprocessing for this task (and perhaps a few others like mutual).

**Review by jeswan — APPROVED** [review]

**Comment by jeswan** [comment]
@yzpang By different form of the dataset, what do you mean? Are there preprocessing scripts run that are not currently in `jiant`?

**Comment by yzpang** [comment]
Right... The preprocessing is complicated. Clara has part of the script and I have part of the script. I am planning on submitting a PR that concatenates the two scripts, and put them into jiant. Bc of this issue, I didn't check the downloader box for quail.

**Review by jeswan — APPROVED** [review]

**Comment by jeswan** [comment]
Remaining work captured in: https://github.com/nyu-mll/jiant/issues/1226

File: jiant/tasks/lib/quail.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return self._create_examples(lines=read_json_lines(self.test_path), set_type="test")
+
+    @classmethod
+    def _create_examples(cls, lines, set_type):
+        examples = []
+        for i, line in enumerate(lines):
+            examples.append(
+                Example(
+                    guid="%s-%s" % (set_type, i),
+                    prompt=line["context"],
```
Comment (HarshTrivedi) [inline jiant/tasks/lib/quail.py:56]: @yzpang @jeswan  It seems that the examples are not using the question at all. Seems strange that the performance still matches the reported ones.

Comment (yzpang) [reply jiant/tasks/lib/quail.py:56]: Interesting. Thanks! I'll check it out soon.

Comment (yzpang) [reply jiant/tasks/lib/quail.py:56]: Addressed in https://github.com/nyu-mll/jiant/pull/1249

**Pull request merged**